packages/shared/src/UI/components/PublicWalletSetting/index.tsx

```diff
@@ -0,0 +1,215 @@
+        alignItems: 'center',
+        borderRadius: 4,
+    },
+    alertTitle: {
+        fontSize: 14,
+        lineHeight: '18px',
+    },
+    placeholder: {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        flex: 1,
+        rowGap: 12,
+    },
+    placeholderText: {
+        color: theme.palette.maskColor.second,
+        fontSize: 14,
+        lineHeight: '18px',
+    },
+    placeholderIcon: {
+        color: theme.palette.maskColor.third,
+    },
+    content: {
+        marginTop: 12,
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: 12,
+        '& > *': {
+            flex: 1,
+            maxWidth: 'calc(50% - 22px)',
+        },
+    },
+}))
+
+interface PublicWalletSettingProps {
+    onClose: () => void
+    onOpenPopup: (route?: PopupRoutes, params?: Record<string, any>) => void
+    bindingWallets?: BindingProof[]
+    currentPersona?: ECKeyIdentifier
+    pluginId: MaskPluginID
+}
+
+export const PublicWalletSetting = memo<PublicWalletSettingProps>(
+    ({ onClose, bindingWallets, currentPersona, pluginId, onOpenPopup }) => {
+        const { classes } = useStyles()
+        const t = useSharedI18N()
+        const { Storage } = useWeb3State()
+
+        const [addresses, setAddresses] = useState<string[]>([])
+        const [showAlert, setShowAlert] = useState(true)
+        const { showSnackbar } = useCustomSnackbar()
+
+        const { value: hiddenAddresses, loading } = useHiddenAddressSetting(pluginId, currentPersona?.publicKeyAsHex)
+
+        const onSwitchChange = useCallback((address: string) => {
+            setAddresses((prev) => {
+                return prev.some((x) => isSameAddress(address, x))
+                    ? prev.filter((x) => !isSameAddress(address, x))
+                    : [...prev, address]
+            })
+        }, [])
+
+        const disabled = useMemo(() => {
+            // If `hiddenAddresses` setting length is different from `addresses`
+            if (hiddenAddresses?.length !== addresses.length) return false
+
+            // If `addresses` is different from setting
+            if (differenceWith(hiddenAddresses ?? EMPTY_LIST, addresses, isSameAddress).length) return false
+
+            return true
+        }, [addresses, hiddenAddresses])
+
+        const [{ loading: confirmLoading }, onConfirm] = useAsyncFn(async () => {
+            try {
+                if (!Storage || !currentPersona) return
+                const storage = Storage.createNextIDStorage(
+                    currentPersona.publicKeyAsHex,
+                    NextIDPlatform.NextID,
+                    currentPersona,
+                )
+                const prevResult = storage.get<PublicWalletSettingType>(MaskPluginID.Web3Profile)
+
+                await storage.set<PublicWalletSettingType>(MaskPluginID.Web3Profile, {
+                    ...prevResult,
+                    hiddenAddresses: addresses,
```

Comment:
Does it possible contains any dup addresses?

```suggestion
                    hiddenAddresses: uniqBy(addresses),
```